bluebutton-data-pipeline-app/src/main/java/gov/hhs/cms/bluebutton/datapipeline/app/S3ToFhirLoadApp.java

```diff
@@ -0,0 +1,245 @@
+package gov.hhs.cms.bluebutton.datapipeline.app;
+
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.net.URISyntaxException;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Slf4jReporter;
+import com.codahale.metrics.jvm.GarbageCollectorMetricSet;
+import com.codahale.metrics.jvm.MemoryUsageGaugeSet;
+
+import gov.hhs.cms.bluebutton.datapipeline.fhir.LoadAppOptions;
+import gov.hhs.cms.bluebutton.datapipeline.fhir.load.FhirBundleResult;
+import gov.hhs.cms.bluebutton.datapipeline.fhir.load.FhirLoader;
+import gov.hhs.cms.bluebutton.datapipeline.fhir.transform.DataTransformer;
+import gov.hhs.cms.bluebutton.datapipeline.fhir.transform.TransformedBundle;
+import gov.hhs.cms.bluebutton.datapipeline.rif.extract.RifFilesProcessor;
+import gov.hhs.cms.bluebutton.datapipeline.rif.extract.s3.DataSetMonitor;
+import gov.hhs.cms.bluebutton.datapipeline.rif.extract.s3.DataSetProcessor;
+import gov.hhs.cms.bluebutton.datapipeline.rif.model.RifFilesEvent;
+import gov.hhs.cms.bluebutton.datapipeline.rif.model.RifRecordEvent;
+
+/**
+ * The main application/driver/entry point for the ETL system, which will pull
+ * any data stored in the specified S3 bucket, transform it, and push it to the
+ * specified FHIR server. See {@link #main(String[])}.
+ */
+public final class S3ToFhirLoadApp {
+	private static final Logger LOGGER = LoggerFactory.getLogger(S3ToFhirLoadApp.class);
+
+	/**
+	 * How often the {@link DataSetMonitor} will wait between scans for new data
+	 * sets.
+	 */
+	private static final Duration S3_SCAN_INTERVAL = Duration.ofSeconds(1L);
```

Comment:
Since loads will happen likely very infrequently, does it make sense to poll much less frequently? Even every 30 seconds or a minute would probably be more than enough.